From 8d1b567224c8278ccbe7d12b1d091fe7d7c0d218 Mon Sep 17 00:00:00 2001
From: "mujiburrahimank@maxlinear.com"
 <mujiburrahimank@lvdocker7.lvsw.maxlinear.com>
Date: Tue, 27 Jan 2026 04:19:00 -0800
Subject: [PATCH] WLANRTSYS-100523 : ACS selecting a fallback channel always
 after bootup.

Fix to avoid aborted scan results which makes faulty ACS and forces it
to select a fallback channel
---
 src/ap/mxl_acs_proprietary.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ap/mxl_acs_proprietary.c b/src/ap/mxl_acs_proprietary.c
index cdba88a3e..2170eb9ff 100644
--- a/src/ap/mxl_acs_proprietary.c
+++ b/src/ap/mxl_acs_proprietary.c
@@ -4859,6 +4859,11 @@ void mxl_event_scan_results(struct hostapd_data *hapd, union wpa_event_data *dat
 		return;
 	}
 
+	if (data->scan_info.aborted) {
+		wpa_printf(MSG_INFO, "ACS: Ignoring aborted scan results to avoid faulty ACS");
+		return;
+	}
+
 	if (hapd->iface->scan_cb)
 		hapd->iface->scan_cb(hapd->iface);
 
-- 
2.43.0

